### PR TITLE
Align help text and labels for tick elements

### PR DESCRIPTION
**Pull request by carkod**
## Done

Align help text with tick elements

Fixes https://github.com/canonical-web-and-design/vanilla-framework/issues/3402
Fixes https://github.com/canonical-web-and-design/vanilla-framework/issues/3966

## QA

- Open https://vanilla-framework-3965.demos.haus/docs/examples/patterns/forms/checkbox
- Open and check help text https://vanilla-framework-3965.demos.haus/docs/examples/patterns/forms/radio
- Check standalon versions also look the same 
https://vanilla-framework-3965.demos.haus/docs/examples/patterns/forms/checkbox
https://vanilla-framework-3965.demos.haus/docs/examples/standalone/patterns/forms/checkbox
https://vanilla-framework-3965.demos.haus/docs/examples/patterns/forms/radio
https://vanilla-framework-3965.demos.haus/docs/examples/standalone/patterns/forms/radio
- Review updated documentation:
  - Added example https://vanilla-framework-3965.demos.haus/docs/base/forms#checkbox

- Also fixed checkbox alignment issue
fix-checkbox-alignement

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical-web-and-design/vanilla-framework/blob/master/templates/docs/whats-new.md).
- [x] Documentation side navigation should be updated with the relevant labels.

### CODE REVIEW

**Comment by lyubomir-popov** [comment]
Hi @carkod the alignment of the secoind line of text looks good, but there's another problem - compare ot the Baseline grid toggle in this screenshot:
image

The checkbox box sits slightly higher in the actual example - is this issue introduced here or present in master as well?

**Comment by bartaz** [comment]
@carkod Did you approve Percy on this PR? There is a lot of unexpected changes in screenshots that shouldn't be approved. Many of checkbox/radio examples are broken because of style changes, such as:

Screenshot 2021-09-06 at 11 37 03

Unfortunately I don't see the option in percy to 'unapprove' them, but you should review all the checkbox related examples and see how they are affected by the change.

**Comment by lyubomir-popov** [comment]
@carkod you should not approve percy yourself, this is another form of visual review that should be done by someone reviewing the pr.

File: scss/_patterns_form-help-text.scss [hunk 1]
```diff
@@ -6,5 +6,10 @@
 
     color: $color-mid-dark;
     margin-top: $sp-unit * 2 - $spv-outer--scaleable; // negate the effect of scaling, as we don't want helper text position to change when the density multiplier changes
+
+    &.is-tick-element {
+      display: block;
+      margin-left: 2rem;
```
Comment (lyubomir-popov) [inline scss/_patterns_form-help-text.scss:12]: what does this 2rem represent? Width of checkbox box + standard spacing in between? If so, thijs would be better:
```suggestion
      margin-left: $sph-inner + $form-tick-box-size;
```

**Comment by lyubomir-popov** [comment]
@carkod @bartaz just push one fix and it will regenerate percy, then please do not approve yourself - just consult it for things that got broken

**Comment by carkod** [comment]
> Hi @carkod the alignment of the secoind line of text looks good, but there's another problem - compare ot the Baseline grid toggle in this screenshot:
> image
> 
> The checkbox box sits slightly higher in the actual example - is this issue introduced here or present in master as well?

Sorry, which checkbox is slightly higher?
What's currently in master is here https://vanillaframework.io/docs/examples/patterns/forms/checkbox

**Comment by lyubomir-popov** [comment]
@carkod 
the box drops a little under the text baseline, so it looks properly centered compared to the text:
image

in this branch, it is aligned to the baseline, making it look a bit raised:
image

When you change from block to inline context, you need to take care of vertical-align to ensure nothing moves.

File: scss/_patterns_form-help-text.scss (hunk 1 above)
Comment (carkod) [reply scss/_patterns_form-help-text.scss:12]: Ok. Thanks. I think it's the `display: block` what's breaking everything, and it's not even needed.

**Comment by carkod** [comment]
> @carkod Did you approve Percy on this PR? There is a lot of unexpected changes in screenshots that shouldn't be approved. Many of checkbox/radio examples are broken because of style changes, such as:
> 
> Screenshot 2021-09-06 at 11 37 03
> 
> Unfortunately I don't see the option in percy to 'unapprove' them, but you should review all the checkbox related examples and see how they are affected by the change.

Fixed.

File: scss/_patterns_strip.scss [hunk 2]
```diff
@@ -108,15 +108,6 @@ $color-suru-slant-left-fallback: rgba(205, 205, 205, 0.14) !default;
       linear-gradient(to bottom left, $color-suru-slant-right-fallback 0%, $color-suru-slant-right-fallback 49.8%, transparent 50%, transparent 100%),
       linear-gradient(to top right, #fff 0%, #fff 49%, transparent 50%, transparent 100%), linear-gradient(to top right, #fff 0%, #fff 100%),
       linear-gradient(111deg, $color-suru-start 10%, $color-suru-middle 37%, $color-suru-end 100%);
-
```
Comment (bartaz) [inline scss/_patterns_strip.scss:111]: @carkod Is this related to this PR at all? It seems it just moves the code around?

Comment (carkod) [reply scss/_patterns_strip.scss:111]: This is the stylelint fix formatting...

File: scss/_patterns_form-tick-elements.scss [hunk 3]
```diff
@@ -1,6 +1,13 @@
 @import 'settings';
 
 @mixin vf-p-form-tick-elements {
+  .p-checkbox,
+  .p-radio {
+    // Fix label alignment when text drops to new line
+    margin-left: $sph-inner + $form-tick-box-size;
```
Comment (bartaz) [inline scss/_patterns_form-tick-elements.scss:7]: It seems that using margin breaks the `u-truncate` utility:

https://vanilla-framework-3965.demos.haus/docs/examples/utilities/truncate
Screenshot 2021-09-07 at 11 43 37

I think using padding instead should work better.

```suggestion
    padding-left: $sph-inner + $form-tick-box-size;
```

File: scss/_patterns_strip.scss (hunk 2 above)
Comment (bartaz) [reply scss/_patterns_strip.scss:111]: @carkod ok, thanks.

Could you remove it from this PR, it's not related and may later be confusing when looking why something was changed.
If you want feel free to push it separately in PR that fixes some of the warnings.

**Review by bartaz — CHANGES_REQUESTED** [review]
Couple small things to fix.

**Comment by bartaz** [comment]
Sorry, still not working, now the "required" asterisk disapeared, probably because of the padding I suggested... I will try to have a look how to fix this.

Screenshot 2021-09-08 at 09 21 49

**Review by bartaz — APPROVED** [review]
LGTM

**Review by carkod — COMMENTED** [review]
LGTM @lyubomir-popov final design review?

**Comment by lyubomir-popov** [comment]
+1'd :)
